# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/TypedArray.js
# Commit: 719bfaded837
# Full Hash: 719bfaded83792edabbeb852a60fe75febf9e65c
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2023-09-21 03:21:32
# Regressor Bug: 1690111
# File Overlap Count: 1
# Description:
#   Bug 1690111 - Implement the ability to pin lengths (prevent changes to length, eg via detaching) for ArrayBuffers and their views. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D131301
# ==============================================================================

diff -r 9497d4dcd737 -r 719bfaded837 js/src/builtin/TypedArray.js
--- a/js/src/builtin/TypedArray.js	Wed Sep 20 09:41:58 2023 +0000
+++ b/js/src/builtin/TypedArray.js	Wed Sep 20 09:41:59 2023 +0000
@@ -9,18 +9,19 @@
 
   var buf = UnsafeGetReservedSlot(tarray, JS_TYPEDARRAYLAYOUT_BUFFER_SLOT);
   assert(
-    buf === null ||
+    buf === false ||
+      buf === true ||
       (IsObject(buf) &&
         (GuardToArrayBuffer(buf) !== null ||
           GuardToSharedArrayBuffer(buf) !== null)),
     "unexpected value in buffer slot"
   );
-  return buf;
+  return IsObject(buf) ? buf : null;
 }
 
 function IsDetachedBuffer(buffer) {
-  // A typed array with a null buffer has never had its buffer exposed to
-  // become detached.
+  // A typed array with a null buffer has never had its buffer exposed,
+  // and so cannot have become detached.
   if (buffer === null) {
     return false;
   }